### PR TITLE
feat(default): colorscheme preview

### PR DESCRIPTION
**Pull request by glepnir**
colorscheme preview when scroll in colorscheme wild menu. restore when press esc cancel.

### CODE REVIEW

**Comment by clason** [comment]
I think that is a bit too intrusive (and niche) to be a default. (Also, I don't set my colors via a color scheme, so your undoing the preview will not work for me.)

File: runtime/lua/vim/_defaults.lua [hunk 1]
```diff
@@ -163,6 +163,30 @@ do
...
+      end
+      local item = vim.split(line, '%s+')[2]
+      if data.event == 'CmdlineLeave' then
+        if not vim.v.event.abort and item then
+          original = item
+        end
+        vim.cmd.color(original)
+      else
+        if not pcall(vim.cmd.colorscheme, { item, mods = { noautocmd = true } }) then
+          vim.cmd.color(original)
```
Comment (justinmk) [inline runtime/lua/vim/_defaults.lua:184]: I'm surprised that `vim.cmd` allows partial command names. That should be an error.

Comment (glepnir) [reply runtime/lua/vim/_defaults.lua:184]: it allows :)

Comment (zeertzjq) [reply runtime/lua/vim/_defaults.lua:184]: Related #20681

**Pull request closed without merging**